### PR TITLE
docs: fix components files

### PR DESCRIPTION
**Pull request by Tanujkanti4441**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
fix eslint component files.

### CODE REVIEW

**Review by nzakas — COMMENTED** [review]
Thanks for the PR, but I'm not sure what this is intended to accomplish. Can you explain the changes you made and why you made them?

File: docs/src/library/code-blocks.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ function() {
     const another = [];
 }
 `` `
+
```
Comment (Tanujkanti4441) [inline docs/src/library/code-blocks.md:21]: added space after the markdown code block as we do when writing examples and it also stated in this page that -
>Make sure to leave space above and below the markdown code block to ensure it is rendered correctly.

File: docs/src/library/rule.md [hunk 2]
```diff
@@ -48,13 +48,13 @@ A rule has a:
 }) }}
 
  {{ rule({
-    name: "getter-return",
+    name: "no-arrow-condition",
     removed: true,
-    description: 'Enforce `return` statements in getters.',
-    replacedBy: "other-rule-here",
```
Comment (Tanujkanti4441) [inline docs/src/library/rule.md:54]: here `other-rule-here` creates multiple links for every word that takes us nowhere.

Screenshot 2024-05-30 164641

File: docs/src/library/rule.md [hunk 3]
```diff
@@ -37,9 +37,9 @@ A rule has a:
 {% from 'components/rule.macro.html' import rule %}
 
  {{ rule({
-    name: "getter-return",
+    name: "array-bracket-newline",
     deprecated: true,
-    description: 'Enforce `return` statements in getters.',
+    description: 'Enforces line breaks after opening and before closing array brackets.',
```
Comment (Tanujkanti4441) [inline docs/src/library/rule.md:42]: here i though to add an example of real deprecated rule.

**Review by nzakas — APPROVED** [review]
Got it, thanks! In the future, please be sure to include details of what you did and why in the PR description.

**Pull request merged**